src/components/token/TokenBalances.js

```diff
@@ -0,0 +1,327 @@
+     */
+    this.state = {
+      tokenId: '00',
+      tokenBalances: [],
+      hasAfter: false,
+      hasBefore: false,
+      sortBy: 'total',
+      order: 'desc',
+      page: 1,
+      pageSearchAfter: [],
+      loading: false,
+      isSearchLoading: false,
+      calculatingPage: false,
+      error: false,
+      tokenBalanceInformationError: false,
+      maintenanceMode: this.props.maintenanceMode,
+      transactionsCount: 0,
+      addressesCount: 0,
+    };
+  }
+
+  componentDidMount = async () => {
+    if (this.state.maintenanceMode) {
+      return;
+    }
+
+    // 'Click' on search to make the first query
+    const queryParams = this.pagination.obtainQueryParams();
+
+    await this.setState({
+      sortBy: get(queryParams, 'sortBy', this.state.sortBy),
+      order: get(queryParams, 'order', this.state.order),
+      loading: true,
+    });
+
+    await this.performSearch();
+
+    this.setState({
+      loading: false,
+    });
+  }
+
+  /**
+   * Call explorer-service to get list of token balances for a given tokenId
+   *
+   * @param {*} searchAfter Parameter needed by ElasticSearch for pagination purposes
+   * @returns tokens
+   */
+  getTokenBalances = async (searchAfter) => {
+    const tokenBalancesRequest = await tokensApi.getBalances(this.state.tokenId, this.state.sortBy, this.state.order, searchAfter);
+    this.setState({
+      error: get(tokenBalancesRequest, 'error', false),
+    });
+    return get(tokenBalancesRequest, 'data', { hits: [], 'has_next': false });
+  }
+
+  loadTokenBalanceInformation = async () => {
+    const tokenBalanceInformationRequest = await tokensApi.getBalanceInformation(this.state.tokenId);
+    this.setState({
+      tokenBalanceInformationError: get(tokenBalanceInformationRequest, 'error', false),
+    });
+
+    return get(tokenBalanceInformationRequest, 'data', {
+      transactions: 0,
+      addresses: 0,
+    });
+  }
+
+  /**
+   * Calls elastic search with state data, sets loading and URL information
+   */
+  performSearch = async () => {
+    this.setState({ isSearchLoading: true });
+    const tokenBalances = await this.getTokenBalances([]);
+    const tokenBalanceInformation = await this.loadTokenBalanceInformation();
+
+    this.setState({
+      isSearchLoading: false,
+      page: 1,
+      tokenBalances: tokenBalances.hits,
+      transactionsCount: tokenBalanceInformation.transactions,
+      addressesCount: tokenBalanceInformation.addresses,
+      hasAfter: tokenBalances.has_next,
+      hasBefore: false,
+      pageSearchAfter: [{
+        page: 1,
+        searchAfter: []
+      }],
+    });
+
+    // This is ultimately called when search text, sort, or sort order changes
+    this.updateURL();
+  }
+
+  /**
+   * Update the URL, so user can share the results of a search
+   */
+  updateURL = () => {
+    const newURL = this.pagination.setURLParameters({
+      sortBy: this.state.sortBy,
+      order: this.state.order,
```

Comment:
Shouldn't we add the search term as a URL parameter as well, to allow sharing the performed search with someone else?

Reply:
I think we can add the selected tokenId, but I suggest doing it in another PR, what do you think?

Reply:
I think in another PR is ok

Reply:
Created an issue for this: https://github.com/HathorNetwork/hathor-explorer/issues/199